lib/smarterEncryption.js

```diff
@@ -130,5 +130,57 @@ function generateSmarterEncryptionRuleset (domains, startingRuleId = 1) {
     return rules
 }
 
+/**
+ * Create a rule to disable or enable automatic HTTPS upgrades for a set of domains.
+ * @param {string[]} domains
+ * @param {number | undefined} id (optional) rule ID
+ * @param {'allow' | 'upgrade'} type 'allow' to create an allowlist rule, 'upgrade' to create an upgrade rule
+ * @returns {{
+ *  rule: import('./utils.js').DNRRule;
+ *  matchDetails: {
+ *      type: string;
+ *      possibleTrackerDomains?: string[];
+ *  };
+ *  }}
+ */
+function createSmarterEncryptionExceptionRule (domains, id = undefined, type = 'allow') {
+    const actionType = type === 'allow' ? 'allow' : 'upgradeScheme'
```

Comment:
How could the exception rules would ever need to have the "upgradeScheme" action? Also, could you add a test for that case?

Reply:
That would be a temporary upgrade rule, used for upgrades via the SE API (see also https://github.com/duckduckgo/duckduckgo-privacy-extension/blob/a48aff60a135f59afbd70b9478fa12ff42be26ad/shared/js/background/https.es6.js#L125).

Reply:
I was thrown off by the naming and thought `Excepiton` implied an allow rule

Reply:
Ah got it. Yea, good idea with that rename I think that's a lot clearer. Could you maybe also expand the JSDoc a little bit for the function to add that context?